lib/models/github-login-model.js

```diff
@@ -1,10 +1,15 @@
+import crypto from 'crypto';
 import {Emitter} from 'event-kit';
 
-import {UNAUTHENTICATED, createStrategy} from '../shared/keytar-strategy';
+import {UNAUTHENTICATED, INSUFFICIENT, createStrategy} from '../shared/keytar-strategy';
 
 let instance = null;
 
 export default class GithubLoginModel {
+  // Be sure that we're requesting at least this many scopes on the token we grant through github.atom.io or we'll
+  // give everyone a really frustrating experience ;-)
+  static REQUIRED_SCOPES = ['repo', 'read:org', 'user:email']
```

Comment:
> unified Atom login experience

I would love to have this so much. We'd need to find a way to make "GitHub identity" a part of the core API somehow, so that packages could just consume it. Maybe have a separate package that exists purely to handle the UI for authentication...

I'm not sure how much effort it would be. We'd need to scope it out first in an Atom RFC and get buy-in from the core team. Maybe we could wait to see how @shana can improve our authentication story first, and then generalize from there.